pkg/util/cluster/cluster.go

```diff
@@ -501,30 +511,68 @@ func (c *Cluster) registerSubscription(ctx context.Context) error {
 		return err
 	}
 
-	req, err := http.NewRequest(http.MethodPut, "https://localhost:8443/subscriptions/"+c.env.SubscriptionID()+"?api-version=2.0", bytes.NewReader(b))
+	req, err := http.NewRequest(http.MethodPut, localDefaultURL+"/subscriptions/"+c.env.SubscriptionID()+"?api-version=2.0", bytes.NewReader(b))
 	if err != nil {
 		return err
 	}
 
 	req.Header.Set("Content-Type", "application/json")
 
-	cli := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		},
-	}
-
-	resp, err := cli.Do(req)
+	resp, err := insecureLocalClient.Do(req)
 	if err != nil {
 		return err
 	}
 
 	return resp.Body.Close()
 }
 
-func (c *Cluster) insertDefaultVersionIntoCosmosdb(ctx context.Context) error {
+// getVersionsInCosmosDB connects to the local RP endpoint and queries the
+// available OpenShiftVersions
+func getVersionsInCosmosDB(ctx context.Context) ([]*api.OpenShiftVersion, error) {
+	type getVersionResponse struct {
+		Value []*api.OpenShiftVersion `json:"value"`
+	}
+
+	getRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, localDefaultURL+"/admin/versions", &bytes.Buffer{})
+	if err != nil {
+		return nil, fmt.Errorf("error creating get versions request: %w", err)
+	}
+
+	getRequest.Header.Set("Content-Type", "application/json")
+
+	getResponse, err := insecureLocalClient.Do(getRequest)
+	if err != nil {
+		return nil, fmt.Errorf("error couldn't retrieve versions in cosmos db: %w", err)
+	}
+
+	parsedResponse := getVersionResponse{}
+	decoder := json.NewDecoder(getResponse.Body)
+	err = decoder.Decode(&parsedResponse)
+
+	return parsedResponse.Value, err
+}
+
+// ensureDefaultVersionInCosmosdb puts a default openshiftversion into the
+// cosmos DB IF it doesn't already contain an entry for the default version. It
+// is hardcoded to use the local-RP endpoint `https://localhost:8443`
+//
+// It returns without an error when a default version is already present or a
+// default version was successfully put into the db.
+func (c *Cluster) ensureDefaultVersionInCosmosdb(ctx context.Context) error {
+	// first, we make sure that we don't overwrite any existing entry for the default version
```

Comment:
Do we need this comment?

Reply:
No worries. I'm borderline anti-comment so I challenge them often. Been bitten by stale comments too many times in the past :D